decryptor/mysql/type_conversion.go

```diff
@@ -62,9 +95,28 @@ func updateFieldEncodedType(field *ColumnDescription, schemaStore config.TableSc
 	if setting := tableSchema.GetColumnEncryptionSettings(string(field.Name)); setting != nil {
 		newFieldType, ok := mapEncryptedTypeToField(setting.GetDBDataTypeID())
 		if ok {
+			fieldConfig, fieldConfigExist := TypeConfigurations[base_mysql.Type(newFieldType)]
+			if !fieldConfigExist {
+				logrus.WithField("field-type", base_mysql.Type(newFieldType)).Debug("No appropriate type configuration")
+				return
+			}
+
 			field.originType = field.Type
 			field.Type = base_mysql.Type(newFieldType)
 			field.changed = true
+
+			field.Charset = fieldConfig.Charset
+			field.ColumnLength = fieldConfig.ColumnLength
+			field.Decimal = fieldConfig.Decimal
+
+			if field.Flag.ContainsFlag(BlobFlag) {
+				for _, fieldType := range specificTypes {
```

Comment:
please, leave a comment why we do it. because variable name `specificTypes` doesn't explain it) and probably put some comment for variable declaration